core/commands/escrow/escrow.go

```diff
@@ -0,0 +1,149 @@
+package escrow
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"time"
+
+	"github.com/TRON-US/go-btfs/core/commands/storage"
+
+	config "github.com/TRON-US/go-btfs-config"
+	"github.com/gogo/protobuf/proto"
+	ic "github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/tron-us/go-btfs-common/crypto"
+	escrowpb "github.com/tron-us/go-btfs-common/protos/escrow"
+	"github.com/tron-us/go-btfs-common/protos/ledger"
+	"google.golang.org/grpc"
+)
+
+func NewContract(configuration *config.Config, id string, payerID string, recverID string, price int64) *escrowpb.EscrowContract {
+	payerAddr := []byte(payerID)
+	recverAddr := []byte(recverID)
+	// TODO: Make escrow GuardAddress configurable
+	//authAddress := configuration.Services.GuardPubKeys[0]
+	authAddress := "guardAddress"
+	return &escrowpb.EscrowContract{
+		ContractId:       []byte(id),
+		BuyerAddress:     payerAddr,
+		SellerAddress:    recverAddr,
+		AuthAddress:      []byte(authAddress),
+		Amount:           price,
+		CollateralAmount: 0,
+		WithholdAmount:   0,
+		TokenType:        escrowpb.TokenType_BTT,
+		PayoutSchedule:   0,
+		NumPayouts:       1,
+	}
+}
+
+func NewSignedContract(contract *escrowpb.EscrowContract) *escrowpb.SignedEscrowContract {
+	return &escrowpb.SignedEscrowContract{
+		Contract: contract,
+	}
+}
+
+func NewContractRequest(configuration *config.Config, chunkInfo map[string]*storage.Chunk) (*escrowpb.EscrowContractRequest, error) {
+	var signedContracts []*escrowpb.SignedEscrowContract
+	var totalPrice int64
+	for _, chunk := range chunkInfo {
+		sc, err := UnmarshalEscrowContract(chunk.SignedContract)
+		if err != nil {
+			return nil, err
+		}
+		signedContracts = append(signedContracts, sc)
+		totalPrice += chunk.Price
+	}
+	// prepare channel commit
+	pid := configuration.Identity.PeerID
+	buyerPrivKey, err := configuration.Identity.DecodePrivateKey("")
+	if err != nil {
+		return nil, err
+	}
+	// TODO: Make escrow Address configurable
+	//escrowAddress := configuration.Services.EscrowPubKeys[0]
+	escrowAddress := "escrowAddress"
+	chanCommit := &ledger.ChannelCommit{
+		Payer:     &ledger.PublicKey{Key: []byte(pid)},
+		Recipient: &ledger.PublicKey{Key: []byte(escrowAddress)},
+		Amount:    totalPrice, // total amount in the contract request
+		PayerId:   time.Now().UnixNano(),
+	}
+	buyerChanSig, err := crypto.Sign(buyerPrivKey, chanCommit)
+	if err != nil {
+		return nil, err
+	}
+
+	signedChanCommit := &ledger.SignedChannelCommit{
+		Channel:   chanCommit,
+		Signature: buyerChanSig,
+	}
+	return &escrowpb.EscrowContractRequest{
+		Contract:     signedContracts,
+		BuyerChannel: signedChanCommit,
+	}, nil
+}
+
+func SubmitContractToEscrow(configuration *config.Config, request *escrowpb.EscrowContractRequest) error {
+	var conn *grpc.ClientConn
+	conn, err := grpc.Dial("52.15.101.94:50051", grpc.WithInsecure())
```

Comment:
this is hard coded IP, need move to env, otherwise expose this ip to public

Reply:
it's pretty much the same as go-btfs-escrow did. And it can be public access right? like ledger address and cert in escorw's config. If using env, local test can be pass, we should call in devopos to add env setting when deployment. So for now, we are not in that stage, we can refine all those to more secure way later on. For testing stage, I would say leave it here.

Reply:
please add a ticket to address this after Robin's general grpc utils is done.

Reply:
> please add a ticket to address this after Robin's general grpc utils is done.

@taiyangc  May be it's better for me to add a ticket and track the grpc common client switch. will do it.

Reply:
ok, please coordinate with shelly.